keps/sig-scheduling/20200612-deducted-topology-manager.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Deducted version of TopologyManager in kube-scheduler
+authors:
+  - "@AlexeyPerevalov"
+owning-sig: sig-scheduling
+participating-sigs:
+reviewers:
+  - "@huang-wei"
+approvers:
+  - "@huang-wei"
+editor: TBD
+creation-date: 2020-05-28
+last-updated: 2020-05-28
+status: implementable
+see-also:
+superseded-by:
+---
+# Topology Aware Scheduling
+
+<!-- toc -->
+- [Summary](#Summary)
+- [Motivation](#Motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#Proposal)
+- [Use cases](#use-cases)
+- [Test plans](#test-plans)
+- [Graduation criteria](#graduation-criteria)
+- [Implementation history](#implementation-history)
+<!-- /toc -->
+
+# Summary
+
+This document describes behavior of the Kubernetes Scheduler which take into
+account worker node NUMA topology.
+
+# Motivation
+
+After Topology Manager was introduced the problem of launching pod in the
+cluster where worker nodes have different NUMA topology and different amount
+of resources in that topology became actual. Pod could be scheduled on the node
+where total amount of resources are enough, but resource distribution could not
+satisfy the appropriate Topology policy. In this case the pod failed to start. Much
+better behaviour for scheduler would be to select appropriate node where admit
+handlers may pass.
+
+
+## Goals
+
+-   Make scheduling process more precise when we have NUMA topology on the
+worker node.
+
+## Non-Goals
+
+-   Do not change Topology Manager behaviour to be able to work with policy in
+the PodSpec
+-   This Proposal requires exposing NUMA topology information. This KEP doesn't
+describe how to expose all necessary information it just declare what kind of
+information is necessary.
```

Comment:
but this is a blocker to having the scheduler work done, so I think we need both KEPs approved at the same time.

Reply:
We tried several approaches to export topology from the worker node, some of them required kubelet modification. And now we came to conclusion what it's better to avoid kubelet modification. There are two feasiable approaches: collect resources on the CRI level and using kubelet podresources interface (unix domain socket, but now it doesn't provide cpumanager information, only resources of the device plugin).
Here I agree, need a detailed description how it would be implemented. Probably KEP should be in sig-node, but as I mentioned before implementation should not touch kubelet.